pkg/apis/minio.min.io/v2/helper.go

```diff
@@ -1139,22 +1139,36 @@ func (t *Tenant) HasConsoleDomains() bool {
 func (t *Tenant) ValidateDomains() error {
 	if t.HasMinIODomains() {
 		domains := t.Spec.Features.Domains.Minio
+		var globalDomains []string
 		if len(domains) != 0 {
-			for _, domainName := range domains {
-				_, err := url.Parse(domainName)
+			for _, domain := range domains {
+				// Infer schema from tenant TLS, if not explicit
+				if !strings.HasPrefix(domain, "http") {
```

Comment:
Can we think of having constants defined for `http` and `https` as they are used in multiple places?

Reply:
Good catch @shtripat ! Based on several discussions (see above), this code will be modified again soon to address how schemas are being used. I will keep this in mind for then.